File: backend/AmfarAPI/Models/Instrumento.cs
namespace AmfarAPI.Models;

public class Instrumento
{
    public int IdInstrumento { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public int StockTotal { get; set; }
    public int StockDisponible { get; set; }
    public string Estado { get; set; } = "Disponible";
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public ICollection<EspecialidadProfesor> EspecialidadProfesores { get; set; } = new List<EspecialidadProfesor>();
    public ICollection<PrestamoInstrumento> Prestamos { get; set; } = new List<PrestamoInstrumento>();
}

File: frontend/AmfarFrontend/src/types/instrumento.ts
export interface InstrumentoResponse {
  idInstrumento: number;
  nombre: string;
  tipo: string;
  stockTotal: number;
  stockDisponible: number;
  estado: string;
  createdAt: string;
}

export interface CrearInstrumentoRequest {
  nombre: string;
  tipo: string;
  stockTotal: number;
}

export interface ActualizarInstrumentoRequest {
  nombre?: string;
  tipo?: string;
  stockTotal?: number;
}

export interface CambiarEstadoInstrumentoRequest {
  estado: string;
}

export interface DisponibilidadResponse {
  idInstrumento: number;
  nombre: string;
  disponible: boolean;
  stockDisponible: number;
}
